### ISSUE TITLE
Feature: Options for flatpickr

### ISSUE THREAD

**Issue by melvinvdb** [ghiscoding/Angular-Slickgrid#142]
**I'm submitting a feature request**

* **Angular Version:**
7.2.7

* **Library Version:**
2.4.1

* **Browser:**
all 
* **Language:**
TypeScript 3.2

**Current/Expected behavior:**
Currently it is not possible to change at least the following:
- `time_24hr`. With this option the time picker will be shown as 24 hour without AM/PM selection which is very handy if the formatted dates are also shown as 24 hour.
- `defaultHour`. The default hour is 12 which means on default you will filter half the day. I would like to set this to 0 to by default filter on the whole day.

**Comment by ghiscoding** [comment]
I assume this is for a Filter (there's also an Editor with Flatpickr), the best would be to add possibility of passing Flatpickr options into the `filterOptions` that is already available as a property under the Column Definition `filter` property. Then use these options in the CompoundDateFilter I assume.

Something like this
```ts
this.columnDefs = [
  { id: 'user', filter: { models: Filters.compoundDate, filterOptions: { yourOptions }} },
];
```

**Comment by melvinvdb** [comment]
Yes that seems to be a nice way to have some control over Flatpickr.

**Comment by ghiscoding** [comment]
Available under the new [version 2.4.2](https://github.com/ghiscoding/Angular-Slickgrid/releases/tag/v2.4.2). 

I made that available in both the Filter (with `filterOptions`) and the Editor (with `editorOptions`). You can see the demo with [filterOptions](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/src/app/examples/grid-clientside.component.ts#L123))
